yaptide/celery/tasks.py

```diff
@@ -0,0 +1,182 @@
+            }
+
+        page: Page   # type is still marked as Never
+        for page in estimator.pages:
+            # currently we are handling for sure only 1-D results
+            # 0-D and 2-D results aren't tested yet, due to testing problems
+
+            # page_dict contains:
+            # "dimensions" indicating it is 1 dim page
+            # "data" which has unit, name and list of data values
+            page_dict = {
+                "dimensions": page.dimension,
+            }
+            # currently output is returned only when dimension == 1 due to
+            # problems in efficient testing of other dimensions
+            if page.dimension == 0:
+                page_dict["data"] = {
+                    "unit": str(page.unit),
+                    "name": str(page.name),
+                    "values": page.data_raw[0],
+                }
+            elif page.dimension == 1:
+                page_dict["data"] = {
+                    "unit": str(page.unit),
+                    "name": str(page.name),
+                    "values": page.data_raw.tolist(),
+                }
+                axis: MeshAxis = page.plot_axis(0)
+                page_dict["first_axis"] = {
+                    "unit": str(axis.unit),
+                    "name": str(axis.name),
+                    "values": axis.data.tolist(),
+                }
+            elif page.dimension == 2:
+                page_dict["data"] = {
+                    "unit": str(page.unit),
+                    "name": str(page.name),
+                    "values": page.data_raw.tolist(),
+                }
+                axis: MeshAxis = page.plot_axis(0)
+                page_dict["first_axis"] = {
+                    "unit": str(axis.unit),
+                    "name": str(axis.name),
+                    "values": axis.data.tolist(),
+                }
+                axis: MeshAxis = page.plot_axis(1)
+                page_dict["second_axis"] = {
+                    "unit": str(axis.unit),
+                    "name": str(axis.name),
+                    "values": axis.data.tolist(),
+                }
+
+            else:
+                # Add info about the location of the file containging to many dimensions
+                raise ValueError(f'Invalid number of pages {page.dimensions}')
+
+            est_dict["pages"].append(page_dict)
+        result_dict["estimators"].append(est_dict)
+
+    return result_dict
+
+
+@celery_app.task
+def simulation_task_status(task_id: str) -> dict:
+    """Task responsible for returning simulation status"""
+    task = AsyncResult(id=task_id, app=celery_app)
+
+    result = {
+        'status': 'OK',
+        'message': {
+            'state': task.state
+        }
+    }
+    if task.state == "PENDING":
+        result['message']['status'] = 'Pending...'
+    elif task.state == "PROGRESS":
+        result['message']['status'] = task.state
+        sim_info = read_shieldlog(path_to_file=task.info.get(
```

Comment:
Add a comment mentioning that this is temporary solution, based on reading logs just from a first directory.
Add a TODO comment and issue to make it better

Reply:
Added in function, because the flow of this code won't change (just function will be)